contracts/liquidityMining/KyberFairLaunchV2.sol

```diff
@@ -176,20 +183,18 @@ contract KyberFairLaunchV2 is IKyberFairLaunchV2, PermissionAdmin, ReentrancyGua
     require(_startTime > _getBlockTime() && _endTime > _startTime, 'add: invalid times');
 
     GeneratedToken _generatedToken;
-    if (_hasGeneratedToken) {
-      _generatedToken = new GeneratedToken(
-        string(abi.encodePacked('KStake - ', GeneratedToken(_stakeToken).name())),
-        string(abi.encodePacked('stk', GeneratedToken(_stakeToken).symbol()))
-      );
+    if (bytes(_tokenName).length != 0 && bytes(_tokenSymbol).length != 0) {
+      _generatedToken = new GeneratedToken(_tokenName, _tokenSymbol);
       poolInfo[poolLength].generatedToken = _generatedToken;
     }
```

Comment:
maybe have an else case that `require(bytes(_tokenName).length == 0 && bytes(_tokenSymbol).length == 0, 'add: bad token name or symbol');`

Reply:
no i think in case of missing token name and symbol then user want to add a pool without generate new token for that pool

Reply:
yeah, so should make sure that both token name and symbol are empty. Ensures that the user really wants to create a pool without token generation.